src/data_collection/yt_data_api.py

```diff
@@ -91,33 +96,39 @@ def get_comments(self, video_data) -> pd.DataFrame:
         comment_count = video_data.comment_count
         unfetched_comments = True
 
-        while unfetched_comments:
-            # The API limits comment requests to 100 records
-            max_comments = min(100, comment_count)
+        with AstroProgress('Downloading comments', comment_count) as progress:
+            while unfetched_comments:
+                # The API limits comment requests to 100 records
+                max_comments = min(100, comment_count)
+
+                self.logger.debug('collecting {} comments'.format(max_comments))
+
+                request = self.youtube.commentThreads().list(
+                    part='snippet,replies',
+                    videoId=video_data.video_id,
+                    pageToken=page_token,
+                    maxResults=max_comments,
+                    textFormat='plainText')
 
-            self.logger.debug('collecting {} comments'.format(max_comments))
+                comment_count -= max_comments
+                unfetched_comments = True if comment_count > 0 else False
 
-            request = self.youtube.commentThreads().list(
-                part='snippet,replies',
-                videoId=video_data.video_id,
-                pageToken=page_token,
-                maxResults=max_comments,
-                textFormat='plainText')
+                try:
+                    response = request.execute()
+                    comment_dataframe, comments_added = self.parse_comment_api_response(response, comment_dataframe)
+                    if 'nextPageToken' in response:  # there are more comments to fetch
+                        page_token = response['nextPageToken']
+                    else:
+                        self.logger.debug("comment collection complete")
+                        unfetched_comments = False
 
-            comment_count -= max_comments
-            unfetched_comments = True if comment_count > 0 else False
+                    progress.advance(comments_added)
 
-            try:
-                response = request.execute()
-                comment_dataframe = self.parse_comment_api_response(response, comment_dataframe)
-                if 'nextPageToken' in response:  # there are more comments to fetch
-                    page_token = response['nextPageToken']
-                else:
-                    unfetched_comments = False
+                except Exception as e:
+                    self.logger.error(str(e))
+                    self.logger.error(traceback.format_exc())
 
-            except Exception as e:
-                self.logger.error(str(e))
-                self.logger.error(traceback.format_exc())
+            progress.complete()
```

Comment:
The comment count retrieved in the video metadata is not always accurate, so it's possible to collect all comments without hitting the `commentCount` returned in the YouTube API response. This can be due to a number of factors which I describe further in the [project wiki](https://github.com/AustinCullar/Astro/wiki/YouTube-Data-API#comment-count).